--- awesome-3.4.13.orig/lib/awful/tag.lua.in
+++ awesome-3.4.13/lib/awful/tag.lua.in
@@ -496,7 +496,7 @@ function attached_add_signal(screen, ...
 end
 
 -- Register standards signals
-capi.client.add_signal("manage", function(c, startup)
+function standart_manage_handler(c, startup)
     -- If we are not managing this application at startup,
     -- move it to the screen where the mouse is.
     -- We only do it for "normal" windows (i.e. no dock, etc).
@@ -514,7 +514,8 @@ capi.client.add_signal("manage", functio
         end
     end
     c:add_signal("property::screen", withcurrent)
-end)
+end
+capi.client.add_signal("manage", standart_manage_handler)
 
 capi.client.add_signal("manage", withcurrent)
 
