--- awesome-3.4.13.orig/screen.c
+++ awesome-3.4.13/screen.c
@@ -61,6 +61,17 @@ screen_default_visual(xcb_screen_t *s)
     return NULL;
 }
 
+int compare_screen_coord(const void *a, const void *b) {
+    const screen_t *sa = (screen_t *)a;
+    const screen_t *sb = (screen_t *)b;
+
+    /* sort order: left to right (x asc), then bottom to top (y desc) */
+
+    int diff = sa->geometry.x - sb->geometry.x;
+
+    return diff ? diff : sb->geometry.y - sb->geometry.y;
+}
+
 /** Get screens informations and fill global configuration.
  */
 void
@@ -113,6 +124,9 @@ screen_scan(void)
                 screen_array_append(&globalconf.screens, s);
             }
         }
+        /* sort irrespective of Xinerama order */
+        qsort(globalconf.screens.tab, globalconf.screens.len, sizeof(screen_t),
+            compare_screen_coord);
 
         p_delete(&xsq);
 
